### PR TITLE
refactor: FF-52 refactor the toggle router

### PR DESCRIPTION
**Pull request by FranLMSP**
#### 🤔 Why?

- Avoid giving the provider the responsibility of getting the toggle context and toggle strategies.

#### 🛠 What I changed:

- A toggle router class was created with the same signature that the toggle provider used to have. And not toggle providers are only responsible of getting the toggle attributes and the list of available toggles.

#### 🗃️ Jira Issues:

- [FF-52](https://ioetec.atlassian.net/browse/FF-52)

[FF-52]: https://ioetec.atlassian.net/browse/FF-52?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ioet_feature_flag/toggles.py [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
 _TOGGLES_LOCATION = './feature_toggles/feature-toggles.yaml'
 
 
 class Toggles:
     def __init__(self, provider: typing.Optional[Provider] = None) -> None:
-        if provider:
-            self._provider = provider
-            return
-        self._provider = YamlToggleProvider(_TOGGLES_LOCATION)
+        if not provider:
```
Comment (Mikkel-14) [inline ioet_feature_flag/toggles.py:15]: I believe that this validation is already been taken into account in the router class

**Review by Mikkel-14 — COMMENTED** [review]
I left a couple of comments

File: ioet_feature_flag/providers/yaml_toggle_provider.py [hunk 2]
```diff
@@ -4,33 +4,42 @@
...
+            environment_toggles: typing.Dict = toggles[self._environment]
+            return list(environment_toggles.keys())
 
-            missing_toogles = [
-                toggle for toggle in toggle_names if toggle not in environment_toggles
-            ]
-            if missing_toogles:
+    def get_toggle_attributes(self, toggle_name: str) -> typing.Dict:
+        with open(self._path, "r") as toggles_file:
+            toggles = yaml.safe_load(toggles_file)
```
[the remark this answers is not in the document]
Comment (FranLMSP) [reply ioet_feature_flag/providers/yaml_toggle_provider.py:23]: yeah good idea, we can apply the template method pattern at some point :p

**Review by Mikkel-14 — APPROVED** [review]
🎏

**Pull request merged**